sway-error/src/warning.rs

```diff
@@ -267,58 +268,52 @@ impl ToDiagnostic for CompileWarning {
                 issue: Issue::warning(
                     source_engine,
                     name.span(),
-                    format!("Constant \"{name}\" should be SCREAMING_SNAKE_CASE"),
+                    format!("Constant \"{name}\" should be SCREAMING_SNAKE_CASE."),
                 ),
                 hints: vec![
-                    Hint::warning(
+                    Hint::help(
                         source_engine,
                         name.span(),
-                        format!("\"{name}\" should be SCREAMING_SNAKE_CASE, like \"{}\".", to_screaming_snake_case(name.as_str())),
+                        format!("Consider renaming it to, e.g., \"{}\".", to_screaming_snake_case(name.as_str())),
                     ),
                 ],
                 help: vec![
                     format!("In Sway, ABIs, structs, traits, and enums are CapitalCase."),
                     format!("Modules, variables, and functions are snake_case, while constants are SCREAMING_SNAKE_CASE."),
-                    format!("Consider renaming the constant to, e.g., \"{}\".", to_screaming_snake_case(name.as_str())),
                 ],
             },
-            MatchExpressionUnreachableArm { match_value, preceding_arms, unreachable_arm, is_last_arm, is_catch_all_arm } => Diagnostic {
+            MatchExpressionUnreachableArm { match_value, match_type, preceding_arms, unreachable_arm, is_last_arm, is_catch_all_arm } => Diagnostic {
                 reason: Some(Reason::new(code(1), "Match arm is unreachable".to_string())),
                 issue: Issue::warning(
                     source_engine,
                     unreachable_arm.clone(),
                     match (*is_last_arm, *is_catch_all_arm) {
-                        (true, true) => format!("Catch-all pattern \"{}\" in the last match arm will never be matched", unreachable_arm.as_str()),
-                        _ => format!("Pattern \"{}\" will never be matched", unreachable_arm.as_str())
+                        (true, true) => format!("Last catch-all match arm `{}` is unreachable.", unreachable_arm.as_str()),
+                        _ => format!("Match arm `{}` is unreachable.", unreachable_arm.as_str())
                     }
                 ),
                 hints: vec![
-                    Hint::warning(
-                        source_engine,
-                        unreachable_arm.clone(),
-                        format!("{} arm \"{}\" is unreachable.", if *is_last_arm && *is_catch_all_arm { "Last catch-all match" } else { "Match" }, unreachable_arm.as_str())
-                    ),
                     Hint::info(
                         source_engine,
                         match_value.clone(),
-                        "This is the value to match on.".to_string()
+                        format!("`{}`, of type \"{match_type}\", is the value to match on.", match_value.as_str())
```

Comment:
This should probably be something like this:
```suggestion
                        format!("`{}`, of type \"{match_type}\", is the expression to match on.", match_value.as_str())
```

Reply:
After a day of pondering, mixed feelings, and a strong urge to start an interesting bike-shedding round :smile:  I've came to conclusion that I don't like that hint formulation in general. It's too long and clumsy for what it wants to say and the value it brings - pointing to the matched expression and stating its type, which is useful.

I am more for something simple like: _"x" has type "(bool, u64)"._

Or in the longer case: _Expression "x" has type "(bool, u64)"._

But let's do that as a part of an upcoming PR which will also be related to match expressions.